src/BenchmarkDotNet/Loggers/AsyncProcessOutputReader.cs

```diff
@@ -103,34 +115,44 @@ private void Detach()
 
         private void ProcessOnOutputDataReceived(object sender, DataReceivedEventArgs e)
         {
-            if (!string.IsNullOrEmpty(e.Data))
+            if (e.Data != null)
             {
-                output.Enqueue(e.Data);
-
-                if (logOutput)
+                if (!string.IsNullOrEmpty(e.Data))
                 {
-                    lock (this) // #2125
+                    output.Enqueue(e.Data);
+
+                    if (logOutput)
                     {
-                        logger.WriteLine(e.Data);
+                        lock (this) // #2125
+                        {
+                            logger.WriteLine(e.Data);
+                        }
                     }
                 }
             }
+            else
```

Comment:
nit: it would be nice to add a comment that `null` means EOF

Reply:
Done.